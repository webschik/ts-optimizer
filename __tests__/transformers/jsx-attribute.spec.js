const ts = require('typescript');
const {createJsxAttributeTransformerFactory} = require('../../src/transformers/jsx-attribute');

describe('Transformers', () => {
    describe('JsxAttribute', () => {
        const printer = ts.createPrinter();

        it('should remove leading and trailing spaces in template literals', () => {
            const inputFile = ts.createSourceFile(
                'test.ts',
                `
                    export default class Button {
                        render () {
                            const className1 = 'className1';
                            const className2 = 'className2';
    
                            return (
                                <button disabled
                                        class={\`   \${className1}  \${className2}  \`}
                                        className={\`
                                            \${className1}
                                            \${\`
                                                \${className2}    
                                            \`}
                                        \`}
                                />
                            );
                        }
                    }
                `,
                ts.ScriptTarget.ESNext,
                true,
                ts.ScriptKind.TSX
            );
            const result = ts.transform(inputFile, [
                createJsxAttributeTransformerFactory({
                    attributesWithTemplateLiterals: ['className', 'class']
                })
            ]);
            const [outputFile] = result.transformed;

            expect(outputFile).toBeDefined();
            const outputFileText = printer.printFile(outputFile);

            expect(printer.printFile(inputFile)).not.toBe(outputFileText);
            expect(outputFileText).toMatchSnapshot();
        });
    });
});