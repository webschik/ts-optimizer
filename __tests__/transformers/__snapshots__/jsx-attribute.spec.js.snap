// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transformers JsxAttribute should remove leading and trailing spaces in template literals 1`] = `
"export default class Button {
    render() {
        const className1 = 'className1';
        const className2 = 'className2';
        return (<button disabled class={\`item \${className1} \${className2} item2\`} className={\`\${className1} \${\`\${className2}\`}\`}/>);
    }
}
"
`;
